### PR TITLE
gadget/install: split Run in several functions

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
Split the Run function that performs the installation in more manageable functions that can be more easily reused.

### CODE REVIEW

File: gadget/install/install.go [hunk 1]
```diff
@@ -239,9 +273,23 @@ func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options
...
 	}
 
+	return allLaidOutVols, created, diskLayout.SectorSize, laidOutBootVol.Name, nil
+}
+
+// Run creates partitions, encrypts them when expected, creates
+// filesystems, and finally writes content on them.
+func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options, observer gadget.ContentObserver, perfTimings timings.Measurer) (*InstalledSystemSideData, error) {
+	// Step 1: create partitions
+	allLaidOutVols, created, sectorSize, bootVolGadgetName, err :=
```
Comment (mvo5) [inline gadget/install/install.go:286]: (nitpick^3) I prefer the longer lines here over spliting into two :)

**Review by mvo5 — APPROVED** [review]
Thank you, this looks fine

File: gadget/install/install.go [hunk 2]
```diff
@@ -163,37 +170,64 @@ func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.En
...
+		return "", nil, err
+	}
+
 	return fsDevice, encryptionKey, nil
 }
 
-// Run bootstraps the partitions of a device, by either creating
-// missing ones or recreating installed ones.
-func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options, observer gadget.ContentObserver, perfTimings timings.Measurer) (*InstalledSystemSideData, error) {
+// Returns laidout volume, on disk structure, sector size, volume name from gadget.yaml
```
Comment (pedronis) [inline gadget/install/install.go:213]: this is not really a proper doc comments, also usually when something returns so many thing we try to use return argument names to clarify things. Though in this case the return arguments are really a bit messy.

File: gadget/install/install.go [hunk 3]
```diff
@@ -163,37 +170,64 @@ func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.En
...
+func createPartitions(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options,
+	perfTimings timings.Measurer) (map[string]*gadget.LaidOutVolume, []gadget.OnDiskStructure, quantity.Size, string, error) {
```
Comment (pedronis) [inline gadget/install/install.go:215]: it seems cleaner to me to follow the patterns of the preexisting in functions an make the return values:

created, laidOutVBoolVol, allLaidOutVols, bootDiskLayout, error

File: gadget/install/install.go (hunk 1 above)
Comment (pedronis) [reply gadget/install/install.go:286]: see comment above about how to change the return values of this

**Review by pedronis — COMMENTED** [review]
bunch of comments

File: gadget/install/install.go [hunk 4]
```diff
@@ -96,7 +96,7 @@ func saveStorageTraits(allLaidOutVols map[string]*gadget.LaidOutVolume, optsPerV
 	return nil
 }
 
-func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.EncryptionType, sectorSize quantity.Size, observer gadget.ContentObserver, perfTimings timings.Measurer) (fsDevice string, encryptionKey keys.EncryptionKey, err error) {
+func encryptPartition(part *gadget.OnDiskStructure, encryptionType secboot.EncryptionType, sectorSize quantity.Size, perfTimings timings.Measurer) (fsDevice string, encryptionKey keys.EncryptionKey, encryptedSectorSize quantity.Size, err error) {
```
[the remark this answers is not in the document]
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Changed to `maybeEncryptPartition`

[the remark this answers is not in the document]
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Done

File: gadget/install/install.go [hunk 5]
```diff
@@ -146,11 +146,18 @@ func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.En
...
-			return "", nil, err
+			return "", nil, 0, err
 		}
 		fsSectorSize = quantity.Size(fsSectorSizeInt)
 	}
 
+	return fsDevice, encryptionKey, fsSectorSize, nil
+}
+
+func createFilesystem(part *gadget.OnDiskStructure, perfTimings timings.Measurer, fsSectorSize quantity.Size, fsDevice string) error {
```
[the remark this answers is not in the document]
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Changed now, thanks for the details.

File: gadget/install/install.go [hunk 6]
```diff
@@ -163,37 +170,64 @@ func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.En
 			})
 		})
 	if err != nil {
-		return "", nil, fmt.Errorf("cannot make filesystem for partition %s: %v", partDisp, err)
+		return fmt.Errorf("cannot make filesystem for partition %s: %v", partDisp, err)
 	}
+	return nil
+}
 
+func writePartitionContent(part *gadget.OnDiskStructure, perfTimings timings.Measurer, observer gadget.ContentObserver, fsDevice string) error {
```
[the remark this answers is not in the document]
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Done

File: gadget/install/install.go (hunk 2 above)
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Change now

File: gadget/install/install.go (hunk 3 above)
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Changed now, although names are not exactly as you wrote them - I've tried to make something similar in spirit.

File: gadget/install/install.go (hunk 1 above)
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Changed now. Re: lines, for me it makes it more readable, but of course it is a matter of opinion,

**Review by alfonsosanchezbeato — COMMENTED** [review]
Thanks for the reviews, I've repushed now.

**Review by pedronis — COMMENTED** [review]
thanks, one naming comment

File: gadget/install/install.go [hunk 7]
```diff
@@ -136,64 +143,96 @@ func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.En
...
 
-// Run bootstraps the partitions of a device, by either creating
-// missing ones or recreating installed ones.
-func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options, observer gadget.ContentObserver, perfTimings timings.Measurer) (*InstalledSystemSideData, error) {
+// createPartitions creates partitions on the disk and returns the
+// volume name where partitions have been created, the on disk
+// structures after that, the laidout volumes, and the disk sector
+// size.
+func createPartitions(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options,
+	perfTimings timings.Measurer) (bootVolGadgetName string, created []gadget.OnDiskStructure, allLaidOutVols map[string]*gadget.LaidOutVolume, sectorSize quantity.Size, err error) {
```
[the remark this answers is not in the document]
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Done now

File: gadget/install/install.go [hunk 8]
```diff
@@ -202,46 +241,61 @@ func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options
...
+	bootVolGadgetName = laidOutBootVol.Name
+	bootVolSectorSize = diskLayout.SectorSize
+	return bootVolGadgetName, created, allLaidOutVols, bootVolSectorSize, nil
+}
+
+// Run creates partitions, encrypts them when expected, creates
+// filesystems, and finally writes content on them.
+func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options, observer gadget.ContentObserver, perfTimings timings.Measurer) (*InstalledSystemSideData, error) {
+	// Step 1: create partitions
+	bootVolGadgetName, created, allLaidOutVols, sectorSize, err :=
```
Comment (pedronis) [inline gadget/install/install.go:292]: I would rename sectorSize here too for clarity

**Review by pedronis — APPROVED** [review]
thanks, one remark

File: gadget/install/install.go (hunk 8 above)
Comment (alfonsosanchezbeato) [reply gadget/install/install.go]: Done

**Pull request merged**